assignments/prototypes.js

```diff
@@ -7,14 +7,24 @@
   
   Each constructor function has unique properties and methods that are defined in their block comments below:
 */
-  
+
 /*
   === GameObject ===
-  * createdAt
```

Comment:
function GameObject(options) {
  this.createdAt = options.createdAt;
  this.dimensions = options.dimensions;
  this.name = options.name;
}

GameObject.prototype.destroy = function () {
  return `${this.name} was removed from the game.`;
};

/*
=== CharacterStats ===
* healthPoints
* name
* takeDamage() // prototype method -> returns the string '<object name> took damage.'
* should inherit destroy() from GameObject's prototype
*/

function CharacterStats(characterStatsOptions) {
  GameObject.call(this, characterStatsOptions);
  this.healthPoints = characterStatsOptions.healthPoints;
}

// Sets up inheritance with GameObject
CharacterStats.prototype = Object.create(GameObject.prototype);

CharacterStats.prototype.takeDamage = function () {
  return `${this.name} took damage.`;
};

/*
=== Humanoid (Having an appearance or character resembling that of a human.) ===
* team
* weapons
* language
* greet() // prototype method -> returns the string '<object name> offers a greeting in <object language>.'
* should inherit destroy() from GameObject through CharacterStats
* should inherit takeDamage() from CharacterStats
*/
function Humanoid(humanoidOptions) {
  CharacterStats.call(this, humanoidOptions);
  this.team = humanoidOptions.team;
  this.weapons = humanoidOptions.weapons;
  this.language = humanoidOptions.language;
}

Humanoid.prototype = Object.create(CharacterStats.prototype);

Humanoid.prototype.greet = function () {
  return `${this.name} offers a greeting in ${this.language}.`;
};